sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/randomExpressions.scala

```diff
@@ -342,6 +343,10 @@ case class RandStr(
   override def left: Expression = length
   override def right: Expression = seedExpression
 
+  override def inputTypes: Seq[AbstractDataType] = Seq(
+    TypeCollection(NumericType),
```

Comment:
```suggestion
    IntegerType,
```
if length is not integer (implicit cast can't be added), analyzer will fail

Reply:
sounds good, done.